source/common/upstream/upstream_impl.cc

```diff
@@ -799,6 +799,43 @@ ClusterInfoImpl::ClusterInfoImpl(
 
     lb_type_ = LoadBalancerType::ClusterProvided;
     break;
+  case envoy::config::cluster::v3::Cluster::LOAD_BALANCING_POLICY_CONFIG: {
+    if (config.has_lb_subset_config()) {
+      throw EnvoyException(
+          fmt::format("cluster: LB policy {} cannot be combined with lb_subset_config",
+                      envoy::config::cluster::v3::Cluster::LbPolicy_Name(config.lb_policy())));
+    }
+
+    if (config.has_common_lb_config()) {
+      throw EnvoyException(
+          fmt::format("cluster: LB policy {} cannot be combined with common_lb_config",
+                      envoy::config::cluster::v3::Cluster::LbPolicy_Name(config.lb_policy())));
+    }
+
+    if (!config.has_load_balancing_policy()) {
+      throw EnvoyException(
+          fmt::format("cluster: LB policy {} requires load_balancing_policy to be set",
+                      envoy::config::cluster::v3::Cluster::LbPolicy_Name(config.lb_policy())));
+    }
+
+    const auto& lb_policy =
+        std::find_if(config.load_balancing_policy().policies().begin(),
+                     config.load_balancing_policy().policies().end(),
+                     [](const envoy::config::core::v3::TypedExtensionConfig& policy) {
+                       return Registry::FactoryRegistry<TypedLoadBalancerFactory>::getFactory(
+                                  policy.name()) != nullptr;
```

Comment:
We no longer use `name` in extension config to identify which factory should be used, it is determined by the type_url in typed_config, so this you should use `Config::Utility::getAndCheckFactory()` and pass the whole message instead.

Reply:
Done, thanks.